src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_validators.py

```diff
@@ -52,22 +65,101 @@ def validate_client_parameters(namespace):
         else:
             raise ValueError("Storage account '{}' not found.".format(n.account_name))
 
-def get_content_setting_validator(settings_class):
+def validate_source_uri(namespace):
+    usage_string = 'invalid usage: supply only one of the following argument sets:' + \
+                   '\n\t   --source-uri' + \
+                   '\n\tOR --source-container --source-blob [--source-snapshot] [--source-sas]' + \
+                   '\n\tOR --source-share --source-path [--source-sas]'
+    ns = vars(namespace)
```

Comment:
Otherwise you get "blahblahblah got unexpected keyword argument blah". In other words, these are convenience parameters we are manually adding to build up the URI (because Xplat has them) so we have to take them back out because they cannot be sent to the SDK call. 

---
In reply to: [77908116](https://github.com/Azure/azure-cli/pull/804#discussion_r77908116) [](ancestors = 77908116,77399658,77393565)